packages/plugin-ext/src/hosted/node/plugin-reader.ts

```diff
@@ -45,14 +46,16 @@ export class HostedPluginReader implements BackendApplicationContribution {
     configure(app: express.Application): void {
         app.get('/hostedPlugin/:pluginId/:path(*)', (req, res) => {
             const pluginId = req.params.pluginId;
-            const filePath = decodeURIComponent(req.params.path);
```

Comment:
> Express seems to already do it.

Could you confirm it?

It was introduced to be aligned with https://github.com/theia-ide/theia/pull/5057/files#diff-2a5766562161792548814f3e490b89ffR55 Since paths to icons are relative, they can contain `.`, `..` and other path elements which have special meaning for browsers it should be encoded. You can find some VS Code extensions using such part for images on https://github.com/theia-ide/theia/pull/5057. And try to test whether express actually decode anything. If not please keep it.

Ideally we should redesign plugin endpoints that they don't try to convert URIs and paths from metadata to URl which are then treated by browser. But just send them as params as is to avoid any modifications to them.

Reply:
btw we have similar issues for webviews: https://github.com/theia-ide/theia/pull/5619

Reply:
ok, found here https://expressjs.com/en/api.html#req.params

> NOTE: Express automatically decodes the values in req.params (using decodeURIComponent).

Please write it down at least a manual test to confirm that relative paths to icons are still working. Please see #5057.

Reply:
I was trying to use [this](https://github.com/tetchel/theia-tree-bug/tree/theia), but for some reason I do not find how to properly use it/get it to work. Do you recommend a specific extension to test icons?

Reply:
@benoitf @evidolob are you able to confirm that this doesn't break anything regarding resource serving?

I tried this [`tree-view-sample`](https://github.com/microsoft/vscode-extension-samples/tree/master/tree-view-sample) extension, but it seems like Theia doesn't accept new views via plugins? I am really not familiar with VS Code extensions, and I don't know how else I could test this PR...

Reply:
I did make requests manually and the files were being served. I simply wasn't able to write a plugin that would somehow make such a request.

Reply:
Maybe https://github.com/microsoft/vscode-extension-samples/tree/master/webview-sample

Reply:
I can see the cat, is this enough in your opinion?

Reply:
yes. Not the cat image directly as it's a remote URL but main.js is served by server so yes